### PR TITLE
Feat: add Grid component

### PR DESCRIPTION
**Pull request by dinartecc**

### CODE REVIEW

**Review by RosaIsela31 — APPROVED** [review]
LGTM! 🚀

File: packages/gatsby-theme/src/components/Grid/Grid.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,139 @@
+import { Meta, Story, Props, Preview } from "@storybook/addon-docs/blocks"
+import { withA11y } from "@storybook/addon-a11y"
```
Comment (omero) [inline packages/gatsby-theme/src/components/Grid/Grid.stories.mdx:2]: Good job, but I don't see any decorator invoking `a11y`, this is expected?
if not we should probably remove it, and adding in a future change :)

**Review by georch — APPROVED** [review]
LGTM

File: packages/gatsby-theme/src/components/Grid/Grid.stories.mdx (hunk 1 above)
Comment (georch) [reply packages/gatsby-theme/src/components/Grid/Grid.stories.mdx:2]: @omero This is expected, `withA11y` is a Storybook addon and this definition will be reflected in the Storybook interface.

**Pull request merged**